packages/adblocker-electron-preload/src/index.ts

```diff
@@ -7,24 +7,20 @@
  */
 
 import { ipcRenderer } from 'electron';
+import { DOMMonitor, IBackgroundCallback } from '@cliqz/adblocker-content';
 
-import {
-  DOMMonitor,
-  IBackgroundCallback,
-  IMessageFromBackground,
-  injectScript,
-} from '@cliqz/adblocker-content';
-
-function getCosmeticsFiltersFirst(): string[] | null {
-  return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
+interface CosmeticFiltersResponse {
+  active: boolean;
+  error?: string;
 }
-function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
-  ipcRenderer.send('get-cosmetic-filters', window.location.href, data);
+
+function injectCosmeticFilters(data?: Omit<IBackgroundCallback, 'lifecycle'>): Promise<CosmeticFiltersResponse> {
+  return ipcRenderer.invoke('inject-cosmetic-filters', window.location.href, data);
```

Comment:
Just a dedicated question besides the core functionality: did you suffer from any problems because of event/handler names without namespacing: e.g. prefixing event names with `adblocker-`. If you think you'll see problems because of event names used in adblocker library, I think prefixing event names is totally fine as it's internal details.

Reply:
We've didn't run into issues with this, but I can totally see people getting confused because of clashing IPC handlers, I'll prefix the event handlers with `@ghostery/adblocker/` to make this less likely.

Reply:
I think that's a great idea. Thanks for the collaboration.